libraries/engage/core/config/theme-config.schema.json

```diff
@@ -0,0 +1,434 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "definitions": {
+    "color": {
+      "type": "string",
+      "pattern": "^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$"
+    },
+    "shadow": {
+      "type": "string",
+      "minLength": 1
+    },
+    "icon": {
+      "type": "string",
+      "minLength": 1,
+      "pattern": "^(<([^>]+)>)$"
+    }
+  },
+  "type": "object",
+  "additionalProperties": false,
```

Comment:
I see, that you are using the `additionalProperties` setting a lot. But this causes failing validation for configurations which contain "abandoned" settings. In e.g. my testshop i've the shape of `variables` and `shadows` differed from the current default config. They just contained some values that are not part of the config anymore.

Deployment of those shops would fail when the schema would be user for validation. Additionally it would not be possible to add new custom colors.

Reply:
Removed the `additionalProperties` flag.